float64 speed
float64 direction